```
$ scons p=android                       
scons: Reading SConscript files ...
Godot Android!!!!! (armv7) (with neon)
scons: done reading SConscript files.
scons: Building targets ...
/home/akien/Projects/godot/android-ndk-r11c//toolchains/llvm/prebuilt/linux-x86_64/bin/clang++ -o platform/android/os_android.os -c -fno-rtti -fno-exceptions -DNO_SAFE_CAST -DFREETYPE_ENABLED -fPIC -DDEBUG_MEMORY_ALLOC -DSCI_NAMESPACE -DENABLE_DEPRECATED -isystem /home/akien/Projects/godot/android-ndk-r11c//platforms/android-14/arch-arm/usr/include -fno-integrated-as -gcc-toolchain /home/akien/Projects/godot/android-ndk-r11c//toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64 -Wno-invalid-command-line-argument -Wno-unused-command-line-argument -fpic -ffunction-sections -funwind-tables -fstack-protector-strong -fvisibility=hidden -fno-strict-aliasing -Wa,--noexecstack -DANDROID -DNO_STATVFS -DGLES2_ENABLED -D__ARM_ARCH_7__ -D__ARM_ARCH_7A__ -march=armv7-a -mfloat-abi=softfp -mfpu=neon -D__ARM_NEON__ -target armv7-none-linux-androideabi -fno-integrated-as -gcc-toolchain /home/akien/Projects/godot/android-ndk-r11c//toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/arm-linux-androideabi/bin -v -O0 -D_DEBUG -UNDEBUG -DDEBUG_ENABLED -DDEBUG_MEMORY_ALLOC -g -fno-limit-debug-info -DANDROID_ENABLED -DUNIX_ENABLED -DNO_FCNTL -DMPC_FIXED_POINT -DGDSCRIPT_ENABLED -DMINIZIP_ENABLED -DXML_ENABLED -Icore -Icore/math -Itools -Idrivers -I. -Iplatform/android -Ithirdparty/freetype -Ithirdparty/freetype/include -Ithirdparty/libpng platform/android/os_android.cpp
clang version 3.8.243773 
Target: armv7-none-linux-android
Thread model: posix
 "/home/akien/Projects/godot/android-ndk-r11c/toolchains/llvm/prebuilt/linux-x86_64/bin/clang++" -cc1 -triple armv7-none-linux-android -S -disable-free -disable-llvm-verifier -main-file-name os_android.cpp -mrelocation-model pic -pic-level 1 -mthread-model posix -mdisable-fp-elim -relaxed-aliasing -fmath-errno -masm-verbose -no-integrated-as -mconstructor-aliases -munwind-tables -fuse-init-array -target-cpu cortex-a8 -target-feature +soft-float-abi -target-feature -fp-only-sp -target-feature -d16 -target-feature +vfp3 -target-feature -fp16 -target-feature -vfp4 -target-feature -fp-armv8 -target-feature +neon -target-feature -crypto -target-abi aapcs-linux -mfloat-abi soft -target-linker-version 2.24 -v -g -dwarf-column-info -ffunction-sections -coverage-file /home/akien/Projects/godot/godot.git/platform/android/os_android.os -resource-dir /home/akien/Projects/godot/android-ndk-r11c/toolchains/llvm/prebuilt/linux-x86_64/bin/../lib64/clang/3.8.243773 -isystem /home/akien/Projects/godot/android-ndk-r11c//platforms/android-14/arch-arm/usr/include -D NO_SAFE_CAST -D FREETYPE_ENABLED -D DEBUG_MEMORY_ALLOC -D SCI_NAMESPACE -D ENABLE_DEPRECATED -D ANDROID -D NO_STATVFS -D GLES2_ENABLED -D __ARM_ARCH_7__ -D __ARM_ARCH_7A__ -D __ARM_NEON__ -D _DEBUG -U NDEBUG -D DEBUG_ENABLED -D DEBUG_MEMORY_ALLOC -D ANDROID_ENABLED -D UNIX_ENABLED -D NO_FCNTL -D MPC_FIXED_POINT -D GDSCRIPT_ENABLED -D MINIZIP_ENABLED -D XML_ENABLED -I core -I core/math -I tools -I drivers -I . -I platform/android -I thirdparty/freetype -I thirdparty/freetype/include -I thirdparty/libpng -internal-isystem /usr/local/include -internal-isystem /home/akien/Projects/godot/android-ndk-r11c/toolchains/llvm/prebuilt/linux-x86_64/bin/../lib64/clang/3.8.243773/include -internal-externc-isystem /include -internal-externc-isystem /usr/include -O0 -Wno-invalid-command-line-argument -Wno-unused-command-line-argument -fdeprecated-macro -fno-dwarf-directory-asm -fdebug-compilation-dir /home/akien/Projects/godot/godot.git -ferror-limit 19 -fmessage-length 189 -fvisibility hidden -fstandalone-debug -femulated-tls -stack-protector 2 -mstackrealign -fno-rtti -fno-signed-char -fobjc-runtime=gcc -fdiagnostics-show-option -fcolor-diagnostics -o /tmp/os_android-6fadf5.s -x c++ platform/android/os_android.cpp
clang -cc1 version 3.8.243773 based upon LLVM 3.6.0svn default target x86_64-unknown-linux
ignoring nonexistent directory "/include"
#include "..." search starts here:
#include <...> search starts here:
 core
 core/math
 tools
 drivers
 .
 platform/android
 thirdparty/freetype
 thirdparty/freetype/include
 thirdparty/libpng
 /home/akien/Projects/godot/android-ndk-r11c//platforms/android-14/arch-arm/usr/include
 /usr/local/include
 /home/akien/Projects/godot/android-ndk-r11c/toolchains/llvm/prebuilt/linux-x86_64/bin/../lib64/clang/3.8.243773/include
 /usr/include
End of search list.
In file included from platform/android/os_android.cpp:29:
In file included from platform/android/os_android.h:32:
In file included from core/os/input.h:32:
core/object.h:522:8: warning: 'this' pointer cannot be null in well-defined C++ code; pointer may be assumed to always convert to true [-Wundefined-bool-conversion]
                if (!this)
                    ~^~~~
core/object.h:537:8: warning: 'this' pointer cannot be null in well-defined C++ code; pointer may be assumed to always convert to true [-Wundefined-bool-conversion]
                if (!this)
                    ~^~~~
2 warnings generated.
 "/..//bin/as" -mfpu=neon -mfloat-abi=softfp -march=armv7-a -mfpu=neon -I core -I core/math -I tools -I drivers -I . -I platform/android -I thirdparty/freetype -I thirdparty/freetype/include -I thirdparty/libpng --noexecstack -o platform/android/os_android.os /tmp/os_android-6fadf5.s
/..//bin/as: unrecognized option '-mfpu=neon'
clang++: error: assembler command failed with exit code 1 (use -v to see invocation)
scons: *** [platform/android/os_android.os] Error 1
scons: building terminated because of errors.
```

--- a/platform/android/detect.py
+++ b/platform/android/detect.py
@@ -24,9 +24,8 @@ def can_build():
 def get_opts():
 
     return [
-        ('ANDROID_NDK_ROOT', 'the path to Android NDK', os.environ.get("ANDROID_NDK_ROOT", 0)),
-        ('NDK_TARGET', 'toolchain to use for the NDK', os.environ.get("NDK_TARGET", "arm-linux-androideabi-4.9")),
-        ('NDK_TARGET_X86', 'toolchain to use for the NDK x86', os.environ.get("NDK_TARGET_X86", "x86-4.9")),
+        ('ANDROID_NDK_ROOT', 'the path to Android NDK',
+         os.environ.get("ANDROID_NDK_ROOT", 0)),
         ('ndk_platform', 'compile for platform: (android-<api> , example: android-14)', "android-14"),
         ('android_arch', 'select compiler architecture: (armv7/armv6/x86)', "armv7"),
         ('android_neon', 'enable neon (armv7 only)', "yes"),
@@ -100,7 +99,6 @@ def configure(env):
         env['android_arch'] = 'armv7'
 
     if env['android_arch'] == 'x86':
-        env['NDK_TARGET'] = env['NDK_TARGET_X86']
         env["x86_libtheora_opt_gcc"] = True
 
     if env['PLATFORM'] == 'win32':
@@ -116,96 +114,129 @@ def configure(env):
 
     if env['android_arch'] == 'x86':
         env.extra_suffix = ".x86" + env.extra_suffix
+        target_subpath = "x86-4.9"
+        abi_subpath = "i686-linux-android"
+        arch_subpath = "x86"
     elif env['android_arch'] == 'armv6':
         env.extra_suffix = ".armv6" + env.extra_suffix
+        target_subpath = "arm-linux-androideabi-4.9"
+        abi_subpath = "arm-linux-androideabi"
+        arch_subpath = "armeabi"
     elif env["android_arch"] == "armv7":
+        target_subpath = "arm-linux-androideabi-4.9"
+        abi_subpath = "arm-linux-androideabi"
+        arch_subpath = "armeabi-v7a"
         if env['android_neon'] == 'yes':
             env.extra_suffix = ".armv7.neon" + env.extra_suffix
         else:
             env.extra_suffix = ".armv7" + env.extra_suffix
 
-    gcc_path = env["ANDROID_NDK_ROOT"] + "/toolchains/" + env["NDK_TARGET"] + "/prebuilt/"
-
     if (sys.platform.startswith("linux")):
         if (platform.machine().endswith('64')):
-            gcc_path = gcc_path + "/linux-x86_64/bin"
+            host_subpath = "linux-x86_64"
         else:
-            gcc_path = gcc_path + "/linux-x86/bin"
+            host_subpath = "linux-x86"
     elif (sys.platform.startswith("darwin")):
-        gcc_path = gcc_path + "/darwin-x86_64/bin"
-        env['SHLINKFLAGS'][1] = '-shared'
-        env['SHLIBSUFFIX'] = '.so'
+        host_subpath = "darwin-x86_64"
     elif (sys.platform.startswith('win')):
         if (platform.machine().endswith('64')):
-            gcc_path = gcc_path + "/windows-x86_64/bin"
+            host_subpath = "windows-x86_64"
         else:
-            gcc_path = gcc_path + "/windows-x86/bin"
+            host_subpath = "windows-x86"
 
-    env['ENV']['PATH'] = gcc_path + ":" + env['ENV']['PATH']
-    if env['android_arch'] == 'x86':
-        env['CC'] = gcc_path + '/i686-linux-android-gcc'
-        env['CXX'] = gcc_path + '/i686-linux-android-g++'
-        env['AR'] = gcc_path + "/i686-linux-android-ar"
-        env['RANLIB'] = gcc_path + "/i686-linux-android-ranlib"
-        env['AS'] = gcc_path + "/i686-linux-android-as"
-    else:
-        env['CC'] = gcc_path + '/arm-linux-androideabi-gcc'
-        env['CXX'] = gcc_path + '/arm-linux-androideabi-g++'
-        env['AR'] = gcc_path + "/arm-linux-androideabi-ar"
-        env['RANLIB'] = gcc_path + "/arm-linux-androideabi-ranlib"
-        env['AS'] = gcc_path + "/arm-linux-androideabi-as"
+    compiler_path = env["ANDROID_NDK_ROOT"] + \
+        "/toolchains/llvm/prebuilt/" + host_subpath + "/bin"
+    gcc_toolchain_path = env["ANDROID_NDK_ROOT"] + \
+        "/toolchains/" + target_subpath + "/prebuilt/" + host_subpath
+    tools_path = gcc_toolchain_path + "/" + abi_subpath + "/bin"
+
+    # For Clang to find NDK tools in preference of those system-wide
+    env.PrependENVPath('PATH', tools_path)
+
+    env['CC'] = compiler_path + '/clang'
+    env['CXX'] = compiler_path + '/clang++'
+    env['AR'] = tools_path + "/ar"
+    env['RANLIB'] = tools_path + "/ranlib"
+    env['AS'] = tools_path + "/as"
 
     if env['android_arch'] == 'x86':
         env['ARCH'] = 'arch-x86'
     else:
         env['ARCH'] = 'arch-arm'
 
-    import string
-    # include path
-    gcc_include = env["ANDROID_NDK_ROOT"] + "/platforms/" + ndk_platform + "/" + env['ARCH'] + "/usr/include"
-    ld_sysroot = env["ANDROID_NDK_ROOT"] + "/platforms/" + ndk_platform + "/" + env['ARCH']
-    # glue_include=env["ANDROID_NDK_ROOT"]+"/sources/android/native_app_glue"
-    ld_path = env["ANDROID_NDK_ROOT"] + "/platforms/" + ndk_platform + "/" + env['ARCH'] + "/usr/lib"
-    env.Append(CPPPATH=[gcc_include])
-#	env['CCFLAGS'] = string.split('-DNO_THREADS -MMD -MP -MF -fpic -ffunction-sections -funwind-tables -fstack-protector -D__ARM_ARCH_5__ -D__ARM_ARCH_5T__ -D__ARM_ARCH_5E__ -D__ARM_ARCH_5TE__  -Wno-psabi -march=armv5te -mtune=xscale -msoft-float  -fno-exceptions -mthumb -fno-strict-aliasing -DANDROID -Wa,--noexecstack -DGLES2_ENABLED ')
+    sysroot = env["ANDROID_NDK_ROOT"] + \
+        "/platforms/" + ndk_platform + "/" + env['ARCH']
+    common_opts = ['-fno-integrated-as', '-gcc-toolchain', gcc_toolchain_path]
+
+    env.Append(CPPFLAGS=["-isystem", sysroot + "/usr/include"])
+    env.Append(CPPFLAGS=string.split(
+        '-Wno-invalid-command-line-argument -Wno-unused-command-line-argument'))
+    env.Append(CPPFLAGS=string.split(
+        '-fpic -ffunction-sections -funwind-tables -fstack-protector-strong -fvisibility=hidden -fno-strict-aliasing -Wa,--noexecstack'))
+    env.Append(CPPFLAGS=string.split('-DANDROID -DNO_STATVFS -DGLES2_ENABLED'))
 
     env['neon_enabled'] = False
     if env['android_arch'] == 'x86':
-        env.Append(CCFLAGS=string.split('-DNO_STATVFS -fpic -ffunction-sections -funwind-tables -fstack-protector -fvisibility=hidden -D__GLIBC__  -Wno-psabi -ftree-vectorize -funsafe-math-optimizations -fno-strict-aliasing -DANDROID -Wa,--noexecstack -DGLES2_ENABLED'))
+        can_vectorize = True
+        target_opts = ['-target', 'i686-none-linux-android']
     elif env["android_arch"] == "armv6":
-        env.Append(CCFLAGS=string.split('-DNO_STATVFS -fpic -ffunction-sections -funwind-tables -fstack-protector -fvisibility=hidden -D__ARM_ARCH_6__ -D__GLIBC__  -Wno-psabi -march=armv6 -mfpu=vfp -mfloat-abi=softfp -funsafe-math-optimizations -fno-strict-aliasing -DANDROID -Wa,--noexecstack -DGLES2_ENABLED'))
+        can_vectorize = False
+        target_opts = ['-target', 'armv6-none-linux-androideabi']
+        env.Append(CPPFLAGS=string.split(
+            '-D__ARM_ARCH_6__ -march=armv6 -mfpu=vfp -mfloat-abi=softfp'))
     elif env["android_arch"] == "armv7":
-        env.Append(CCFLAGS=string.split('-DNO_STATVFS -fpic -ffunction-sections -funwind-tables -fstack-protector -fvisibility=hidden -D__ARM_ARCH_7__ -D__ARM_ARCH_7A__ -D__GLIBC__  -Wno-psabi -march=armv7-a -mfloat-abi=softfp -ftree-vectorize -funsafe-math-optimizations -fno-strict-aliasing -DANDROID -Wa,--noexecstack -DGLES2_ENABLED'))
+        can_vectorize = True
+        target_opts = ['-target', 'armv7-none-linux-androideabi']
+        env.Append(CPPFLAGS=string.split(
+            '-D__ARM_ARCH_7__ -D__ARM_ARCH_7A__ -march=armv7-a -mfloat-abi=softfp'))
         if env['android_neon'] == 'yes':
             env['neon_enabled'] = True
-            env.Append(CCFLAGS=['-mfpu=neon', '-D__ARM_NEON__'])
+            env.Append(CPPFLAGS=['-mfpu=neon', '-D__ARM_NEON__'])
         else:
-            env.Append(CCFLAGS=['-mfpu=vfpv3-d16'])
+            env.Append(CPPFLAGS=['-mfpu=vfpv3-d16'])
 
-    env.Append(LDPATH=[ld_path])
+    env.Append(CPPFLAGS=target_opts)
+    env.Append(CPPFLAGS=common_opts)
+
     env.Append(LIBS=['OpenSLES'])
-#	env.Append(LIBS=['c','m','stdc++','log','EGL','GLESv1_CM','GLESv2','OpenSLES','supc++','android'])
     env.Append(LIBS=['EGL', 'OpenSLES', 'android'])
-    env.Append(LIBS=['c', 'm', 'stdc++', 'log', 'GLESv1_CM', 'GLESv2', 'z'])
+    env.Append(LIBS=['log', 'GLESv1_CM', 'GLESv2', 'z'])
 
-    env["LINKFLAGS"] = string.split(" -g --sysroot=" + ld_sysroot + " -Wl,--no-undefined -Wl,-z,noexecstack ")
-    env.Append(LINKFLAGS=["-Wl,-soname,libgodot_android.so"])
+    if (sys.platform.startswith("darwin")):
+        env['SHLIBSUFFIX'] = '.so'
 
-    if (env["target"] == "release"):
+    env['LINKFLAGS'] = ['-shared', '--sysroot=' +
+                        sysroot, '-Wl,--warn-shared-textrel',
+                        '-Wl,--threads']
+    env.Append(LINKFLAGS=string.split(
+        '-Wl,--fix-cortex-a8'))
+    env.Append(LINKFLAGS=string.split(
+        '-Wl,--no-undefined -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now'))
+    env.Append(LINKFLAGS=string.split(
+        '-Wl,-soname,libgodot_android.so -Wl,--gc-sections'))
+    env.Append(LINKFLAGS=target_opts)
+    env.Append(LINKFLAGS=common_opts)
 
-        env.Append(CCFLAGS=['-O2', '-ffast-math', '-fomit-frame-pointer'])
+    env.Append(LIBPATH=[env["ANDROID_NDK_ROOT"] + '/toolchains/arm-linux-androideabi-4.9/prebuilt/' +
+                        host_subpath + '/lib/gcc/' + abi_subpath + '/4.9.x'])
+    env.Append(LIBPATH=[env["ANDROID_NDK_ROOT"] +
+                        '/toolchains/arm-linux-androideabi-4.9/prebuilt/' + host_subpath + '/' + abi_subpath + '/lib'])
 
-    elif (env["target"] == "release_debug"):
-
-        env.Append(CCFLAGS=['-O2', '-ffast-math', '-DDEBUG_ENABLED'])
-
+    if (env["target"].startswith("release")):
+        env.Append(LINKFLAGS=['-O2'])
+        env.Append(CPPFLAGS=['-O2', '-DNDEBUG', '-ffast-math',
+                             '-funsafe-math-optimizations', '-fomit-frame-pointer'])
+        if (can_vectorize):
+            env.Append(CPPFLAGS=['-ftree-vectorize'])
+        if (env["target"] == "release_debug"):
+            env.Append(CPPFLAGS=['-DDEBUG_ENABLED'])
     elif (env["target"] == "debug"):
+        env.Append(LINKFLAGS=['-O0'])
+        env.Append(CPPFLAGS=['-O0', '-D_DEBUG', '-UNDEBUG', '-DDEBUG_ENABLED',
+                             '-DDEBUG_MEMORY_ALLOC', '-g', '-fno-limit-debug-info'])
 
-        env.Append(CCFLAGS=['-D_DEBUG', '-g1', '-Wall', '-O0', '-DDEBUG_ENABLED'])
-        env.Append(CPPFLAGS=['-DDEBUG_MEMORY_ALLOC'])
-
-    env.Append(CPPFLAGS=['-DANDROID_ENABLED', '-DUNIX_ENABLED', '-DNO_FCNTL', '-DMPC_FIXED_POINT'])
-#	env.Append(CPPFLAGS=['-DANDROID_ENABLED', '-DUNIX_ENABLED','-DMPC_FIXED_POINT'])
+    env.Append(CPPFLAGS=['-DANDROID_ENABLED',
+                         '-DUNIX_ENABLED', '-DNO_FCNTL', '-DMPC_FIXED_POINT'])
 
     # TODO: Move that to opus module's config
     if("module_opus_enabled" in env and env["module_opus_enabled"] != "no"):
@@ -214,40 +245,22 @@ def configure(env):
         env.opus_fixed_point = "yes"
 
     if (env['android_stl'] == 'yes'):
-        # env.Append(CCFLAGS=[env["ANDROID_NDK_ROOT"]+"/sources/cxx-stl/system/include"])
-        env.Append(CPPPATH=[env["ANDROID_NDK_ROOT"] + "/sources/cxx-stl/gnu-libstdc++/4.9/include"])
-        if env['android_arch'] == 'x86':
-            env.Append(CPPPATH=[env["ANDROID_NDK_ROOT"] + "/sources/cxx-stl/gnu-libstdc++/4.9/libs/x86/include"])
-            env.Append(LIBPATH=[env["ANDROID_NDK_ROOT"] + "/sources/cxx-stl/gnu-libstdc++/4.9/libs/x86"])
-        elif env['android_arch'] == 'armv6':
-            env.Append(CPPPATH=[env["ANDROID_NDK_ROOT"] + "/sources/cxx-stl/gnu-libstdc++/4.9/libs/armeabi/include"])
-            env.Append(LIBPATH=[env["ANDROID_NDK_ROOT"] + "/sources/cxx-stl/gnu-libstdc++/4.9/libs/armeabi"])
-        elif env["android_arch"] == "armv7":
-            env.Append(CPPPATH=[env["ANDROID_NDK_ROOT"] + "/sources/cxx-stl/gnu-libstdc++/4.9/libs/armeabi-v7a/include"])
-            env.Append(LIBPATH=[env["ANDROID_NDK_ROOT"] + "/sources/cxx-stl/gnu-libstdc++/4.9/libs/armeabi-v7a"])
-
-        env.Append(LIBS=["gnustl_static", "supc++"])
-        env.Append(CPPPATH=[env["ANDROID_NDK_ROOT"] + "/sources/cpufeatures"])
-
-        # env.Append(CCFLAGS=["-I"+env["ANDROID_NDK_ROOT"]+"/sources/cxx-stl/stlport/stlport"])
-        # env.Append(CCFLAGS=["-I"+env["ANDROID_NDK_ROOT"]+"/sources/cxx-stl/gnu-libstdc++/libs/armeabi/include"])
-        # env.Append(LINKFLAGS=[env["ANDROID_NDK_ROOT"]+"/sources/cxx-stl/gnu-libstdc++/libs/armeabi/libstdc++.a"])
+        env.Append(CPPPATH=[env["ANDROID_NDK_ROOT"] +
+                            "/sources/cxx-stl/gnu-libstdc++/4.9/include"])
+        env.Append(CPPPATH=[env["ANDROID_NDK_ROOT"] +
+                            "/sources/cxx-stl/gnu-libstdc++/4.9/libs/" + arch_subpath + "/include"])
+        env.Append(LIBPATH=[env["ANDROID_NDK_ROOT"] +
+                            "/sources/cxx-stl/gnu-libstdc++/4.9/libs/" + arch_subpath])
+        env.Append(LIBS=["gnustl_static"])
     else:
-
-        env.Append(CPPPATH=[env["ANDROID_NDK_ROOT"] + "/sources/cxx-stl/gnu-libstdc++/4.9/include"])
-        env.Append(CPPPATH=[env["ANDROID_NDK_ROOT"] + "/sources/cpufeatures"])
-        if env['android_arch'] == 'x86':
-            env.Append(LIBPATH=[env["ANDROID_NDK_ROOT"] + "/sources/cxx-stl/gnu-libstdc++/4.9/libs/x86"])
-        elif env["android_arch"] == "armv6":
-            env.Append(LIBPATH=[env["ANDROID_NDK_ROOT"] + "/sources/cxx-stl/gnu-libstdc++/4.9/libs/armeabi"])
-        elif env["android_arch"] == "armv7":
-            env.Append(LIBPATH=[env["ANDROID_NDK_ROOT"] + "/sources/cxx-stl/gnu-libstdc++/4.9/libs/armeabi-v7a"])
-        env.Append(LIBS=['gnustl_static'])
-        env.Append(CCFLAGS=["-fno-exceptions", '-DNO_SAFE_CAST'])
+        env.Append(CXXFLAGS=['-fno-rtti', '-fno-exceptions', '-DNO_SAFE_CAST'])
 
     import methods
-    env.Append(BUILDERS={'GLSL120': env.Builder(action=methods.build_legacygl_headers, suffix='glsl.h', src_suffix='.glsl')})
-    env.Append(BUILDERS={'GLSL': env.Builder(action=methods.build_glsl_headers, suffix='glsl.h', src_suffix='.glsl')})
-    env.Append(BUILDERS={'GLSL120GLES': env.Builder(action=methods.build_gles2_headers, suffix='glsl.h', src_suffix='.glsl')})
+    env.Append(BUILDERS={'GLSL120': env.Builder(
+        action=methods.build_legacygl_headers, suffix='glsl.h', src_suffix='.glsl')})
+    env.Append(BUILDERS={'GLSL': env.Builder(
+        action=methods.build_glsl_headers, suffix='glsl.h', src_suffix='.glsl')})
+    env.Append(BUILDERS={'GLSL120GLES': env.Builder(
+        action=methods.build_gles2_headers, suffix='glsl.h', src_suffix='.glsl')})
 
     env.use_windows_spawn_fix()

